airbyte-integrations/bases/base-singer/base_singer/source.py

```diff
@@ -22,15 +22,38 @@
 SOFTWARE.
 """
 
+import json
+from dataclasses import dataclass
 from typing import Dict, Generator, Type
 
 from airbyte_protocol import AirbyteCatalog, AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, Status
-from base_python import AirbyteLogger, CatalogHelper, ConfigContainer, Source
+from base_python import AirbyteLogger, CatalogHelper, Source
 
 from .singer_helpers import Catalogs, SingerHelper, SyncModeInfo
 
 
+@dataclass
+class ConfigContainer:
+    rendered_config: object
+    rendered_config_path: str
+
+
 class SingerSource(Source):
+    # can be overridden to change an input file config
+    def transform_config(self, config_path: str, config: json):
```

Comment:
does this need to returned a ConfigContainer? can it just return the new rendered config and then the internals of the SingerSource handle writing it to disk and tracking the path?